src/main/java/stream/flarebot/flarebot/commands/moderation/PinCommand.java

```diff
@@ -18,10 +18,24 @@
     @Override
     public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Message message, String[] args, Member member) {
         if (args.length == 1 && args[0].matches("[0-9]{18,22}")) {
+            if (!guild.getGuild().getSelfMember().hasPermission(channel, Permission.MESSAGE_MANAGE)) {
```

Comment:
Cam we combine the check for this

Reply:
We don't have the error message then and that's just weird.

Reply:
Move the error message with the check. they error messages are bolth the same. as far as i can tell this is repeat code

Reply:
Move the error message with the check then. the error messages are both the same. as far as i can tell this is repeat code

Reply:
Argument error is different to perm error, repeated code doesn't really apply here as making a method for a 1 line check is useless

Reply:
I wasn't saying make i method i was saying we should combine this and the other if that i mentioned below before the big if so we don't have the same code in two places

Reply:
sorry if that was confusing but thats what i ment by combine them is we have two different if staments that do the same thing

Reply:
Now that I'm looking at it the current way if fine

Reply:
No idea what the issue was anyway